### PR TITLE
Andrew McLaughlin ( Theory-Of-Computation)

### PR DESCRIPTION
**Pull request by LaikaFusion**

### CODE REVIEW

File: project/state_machines/state_machines.c [hunk 1]
```diff
@@ -164,16 +209,37 @@ Transition *sm_add_transition(StateMachine *sm, char *transition_name,
...
+  if(origin_state != NULL && destination_state != NULL){
+     Transition *transition = create_transition(transition_name,origin_state,destination_state);
+     sm->transitions[sm->num_transitions] = transition;
+  sm->num_transitions += 1;
+     return transition;
 
+  }
+  else{
+    printf("Something went wrong\n");
+    return NULL;
```
Comment (kelfro) [inline project/state_machines/state_machines.c]: When I run your code I get a lot of something went wrongs - I am wondering where the error might be. This part of your code look okay but above around line 212, the solution code says that if the capacity is <= number of transitions then print the error and return null. Yours has the capacity and number of transitions being equal and I wonder if that might be affecting the result??

File: project/state_machines/state_machines.c [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 200809L
```
Comment (kelfro) [inline project/state_machines/state_machines.c]: Good job finding this! A lot of people had issues with strdup without it.

Comment (LaikaFusion) [reply project/state_machines/state_machines.c]: That was all Jun, his thread on the cs12_help channel yesterday was great

File: project/state_machines/state_machines.c (hunk 1 above)
Comment (LaikaFusion) [reply project/state_machines/state_machines.c]: I've changed it so hopefully that fixes it.

I also tested the program with: https://www.onlinegdb.com/online_c_compiler It ran fine there

Then I got curious and tried clang. It worked there too.

And then I tried running that at different levels (All the way back to c90). All of those worked except c90 (it doesn't like ints declared in for loops).

Might be time to upgrade from GCC 4.2.1?

**Comment by kelfro** [comment]
You will be pleased to hear that I did upgrade my compiler and it runs in C99 mode now. I think the problem is that I was lazy this morning and didn't want to fire up my other computer so I ran it through a remote server. I was also too silly to use python tutor, which I will try next time my compiler gets angry. :)

**Pull request closed without merging**